server/routes/CommentRoutes.js

```diff
@@ -1,19 +1,23 @@
 const Comment = require("../controllers/CommentController");
 const router = require("express").Router();
+
+const checkRole = require('./middleware/checkRole');
+
 const { AsyncWrapController } = require("../routes/ErrorHandlingMiddleware");
 
 AsyncWrapController(Comment);
 
+
 // Get a comment for a solution.
 router.get("/:solutionId", Comment.get);
 
 // Create a new comment for a solution in the database.
-router.post("/:id", Comment.create);
+router.post("/:id", checkRole(['user']), Comment.create);
```

Comment:
wouldn't it be easier to add middleware in the index file?

Reply:
I think it's better if we have this separation. we might also consider having it more organized with a middleware directory